### ISSUE TITLE
Any reason why mailHistoryRepository is private in MailService

### ISSUE THREAD

**Issue by bravemaster619** [EC-CUBE/ec-cube#4692]
## 概要(Overview)
https://github.com/EC-CUBE/ec-cube/blob/dc27cb56697f2a02a7950d0a12cdc9983aa0d6a3/src/Eccube/Service/MailService.php#L47

` \Eccube\Service\MailService` cannot be extended easily. Because `mailHistoryRepository` has private access, while all the other params has protected access.

## 期待する内容(Expect)  or 要望 (Requirement)
`mailHistoryRepository` should have protected access

## 再現手順(Procedure)
```php
class MailService extends \Eccube\Service\MailService {
    /**
     * {@inheritdoc}
     */
    public function sendOrderMail(Order $Order) {
       parent::sendOrderMail($Order);
       $mailHistory = new MailHistory();
       $this->mailHistoryRepository->save($mailHistory); // will throw exception
    }
}
```

### 環境 (environment)
+ EC-CUBE: latest
+ PHP: any
+ DB:
  - PostgreSQL any
  - MySQL any

**Comment by okazy** [comment]
I don't know why it's private either.
It's limited, so I think we can change it to protected.

**Comment by okazy** [comment]
@bravemaster619 
Thank you for your suggestion :)
This Issue has been corrected to protected in #4693 and I will close this.
